Set up module for rest webservices calls

paulupendo opened the pull request:
# JIRA TICKET NAME:
[OWA-33](https://issues.openmrs.org/browse/OWA-33)

# Summary:
- Adds rest webservies urls to rest controller
- Add rest dependencies to pom xml file

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java:
```diff
@@ -54,7 +55,7 @@
 	// REST implementation
 	// -------------------------------------------------------------------------
-	@RequestMapping(value = "/rest/owa/applist", method = RequestMethod.GET)
+    @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/owa/applist", method = RequestMethod.GET)
```
Just create a new controller and leave this one un touched because we want to preserve existing functionality in order not to break clients that may already be using it.

paulupendo replied:
Okay @dkayiwa. I think implementation remains the same.

paulupendo commented:
@dkayiwa Done.I have followed the documentation to the later.

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaWebservicesRestController.java:
```diff
@@ -0,0 +1,133 @@
+        List<App> appList = new ArrayList<>();
+        if(Context.hasPrivilege("Manage OWA")){
+            String message;
+            HttpSession session = request.getSession();
```
Do we need to duplicate this code?

dkayiwa commented:
The code in your controller is all duplicated. Do we need to do so?

paulupendo commented:
@dkayiwa Is it better now?

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaWebservicesRestController.java:
```diff
@@ -0,0 +1,80 @@
+	@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/owa/applist", method = RequestMethod.GET)
+	@ResponseBody
+	public void getApps() {
+		if (Context.hasPrivilege("Manage OWA")) {
```
Isn't owaRestController already doing the privilege check?

paulupendo commented:
Yes it will but without doing that tests fail with error "failed to get app
Context"

On 6 Sep 2017 6:22 pm, "dkayiwa" <notifications@github.com> wrote:

> *@dkayiwa* commented on this pull request.
> ------------------------------
>
> In omod/src/main/java/org/openmrs/module/owa/web/controller/
> OwaWebservicesRestController.java
> <https://github.com/openmrs/openmrs-module-owa/pull/52#discussion_r137300763>
> :
>
> > +	// -------------------------------------------------------------------------
> +	@Autowired
> +	private AppManager appManager;
> +	
> +	OwaRestController owaRestController;
> +	
> +	@Autowired
> +	private MessageSourceService messageSourceService;
> +	
> +	// -------------------------------------------------------------------------
> +	// REST implementation
> +	// -------------------------------------------------------------------------
> +	@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/owa/applist", method = RequestMethod.GET)
> +	@ResponseBody
> +	public void getApps() {
> +		if (Context.hasPrivilege("Manage OWA")) {
>
> Isn't owaRestController already doing the privilege check?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openmrs/openmrs-module-owa/pull/52#pullrequestreview-60949515>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ASShmIygFnoa_72EoQkdW8cwSM1e3OoZks5sfrjLgaJpZM4PORXY>
> .
>

paulupendo commented:
I can take a look at it tomorrow and see how to better the current
implementation. But I would love for your advice on how to go about it.

On 6 Sep 2017 6:40 pm, "Paul Upendo" <paul.upendo@andela.com> wrote:

> Yes it will but without doing that tests fail with error "failed to get
> app Context"
>
> On 6 Sep 2017 6:22 pm, "dkayiwa" <notifications@github.com> wrote:
>
>> *@dkayiwa* commented on this pull request.
>> ------------------------------
>>
>> In omod/src/main/java/org/openmrs/module/owa/web/controller/Owa
>> WebservicesRestController.java
>> <https://github.com/openmrs/openmrs-module-owa/pull/52#discussion_r137300763>
>> :
>>
>> > +	// -------------------------------------------------------------------------
>> +	@Autowired
>> +	private AppManager appManager;
>> +	
>> +	OwaRestController owaRestController;
>> +	
>> +	@Autowired
>> +	private MessageSourceService messageSourceService;
>> +	
>> +	// -------------------------------------------------------------------------
>> +	// REST implementation
>> +	// -------------------------------------------------------------------------
>> +	@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/owa/applist", method = RequestMethod.GET)
>> +	@ResponseBody
>> +	public void getApps() {
>> +		if (Context.hasPrivilege("Manage OWA")) {
>>
>> Isn't owaRestController already doing the privilege check?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/openmrs/openmrs-module-owa/pull/52#pullrequestreview-60949515>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ASShmIygFnoa_72EoQkdW8cwSM1e3OoZks5sfrjLgaJpZM4PORXY>
>> .
>>
>

dkayiwa commented:
@paulupendo can you do it as part of this pull request and i look at the error the you get?

dkayiwa commented:
@paulupendo these are the controllers am talking about:

https://github.com/openmrs/openmrs-module-appointmentscheduling/blob/master/omod/src/main/java/org/openmrs/module/appointmentscheduling/rest/controller/AppointmentRestController.java

https://github.com/openmrs/openmrs-module-reportingrest/blob/master/omod/src/main/java/org/openmrs/module/reportingrest/web/controller/ReportingRestController.java

paulupendo commented:
@dkayiwa As per the controller examples. Let me rewrite what I have.

dkayiwa commented:
@paulupendo if you want to see more, here we go: https://github.com/search?l=&p=1&q=org%3Aopenmrs+getNamespace%28%29&ref=advsearch&type=Code

paulupendo commented:
@dkayiwa I have rewritten as per OpenMRS REST WS controller specs.

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java:
```diff
@@ -64,7 +64,7 @@
 	        }
 	        return appList;
 		}
-	
```
I thought you would not touch this file at all.

paulupendo replied:
I reverted it to what it was before after making changes earlier. But let me take it back to what it was on the master branch.

Commit:
[Chore]Add webservicesRest Dependency

[Chore]Create AddOnManager webservices.rest Controller

[Chore]Implement deprecation

Delete previous controller

[Chore]use proper javadoc

[Chore]use proper javadoc

Rename namespace variable

paulupendo commented:
@dkayiwa I have reverted changes and made them as requested. Thank you for the guidance. If this is up to standard. What are the next steps?

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/AddOnManagerRestController.java:
```diff
@@ -0,0 +1,21 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+
```
Can you deprecate OwaRestController and also put a comment here that once OwaRestController is deleted, this class should be renamed to OwaRestController? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation

dkayiwa replied:
And for now i would call it OwaManagerRestController.java

paulupendo commented:
@dkayiwa I have done the deprecation as requested.

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaManagerRestController.java:
```diff
@@ -0,0 +1,22 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+// Once OwaRestController is deleted, this class should be renamed to OwaRestController
```
Why not do the above as a proper class javadoc?

paulupendo commented:
@dkayiwa Done. I have changed them to proper javadoc comments. Thank you for pointing this out.

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaManagerRestController.java:
```diff
@@ -0,0 +1,24 @@
+/**
+ * Once OwaRestController is deleted, this class should be renamed to OwaRestController
+ */
+
```
Can we remove the above space?

paulupendo commented:
@dkayiwa Done 🙂

dkayiwa commented on omod/src/main/java/org/openmrs/module/owa/web/controller/OwaManagerRestController.java:
```diff
@@ -0,0 +1,24 @@
+@Controller
+@RequestMapping("/rest/" + RestConstants.VERSION_1 + OwaManagerRestController.ADD_ON_MANAGER_NAMESPACE)
+public class OwaManagerRestController extends MainResourceController {
+    public static final String ADD_ON_MANAGER_NAMESPACE = "/owa";
```
And lastly, this would be OWA_NAMESPACE

dkayiwa commented:
@paulupendo can we update the ticket title and description to reflect what has actually been done in this pull request?

paulupendo commented:
@dkayiwa Done as requested
